tests/test_security_curve.cpp

```diff
@@ -40,6 +40,10 @@
 #  include <unistd.h>
 #endif
 
+#include "../src/tweetnacl.h"
```

Comment:
Yes, the code is leaking. I did not deny that ;)
What do you think about putting such code into a separate library that both libzmq and specific tests link against?
I am very much against duplicating code, even for things that appear trivial. And the code implementing the CURVE protocol is not trivial at all.